util.py

```diff
@@ -107,25 +109,48 @@ def _cli():
     return bool(ctx().args and hasattr(ctx().args, '._cli') and ctx().args._cli)
 
 
+# A collection of string-identifiers of terminals capable of rendering colours
+FORMATTING_COMPATIBLE_TERMINALS = ['xterm']
+
+
+def build_message(category: str, message: str, colour: str='white', style: str='bold'):
```

Comment:
this can be done more elegant:

evaluate the terminal type ONCE; based on whether or not colouring is supported, set the formatting function to either "print" or the custom one